concrete-npe/src/operators.rs

```diff
@@ -0,0 +1,731 @@
+/// let base_log = DecompositionBaseLog(7);
+/// let dispersion_lwe = Variance(f64::powi(2., -38));
+/// let dispersion_ks = Variance(f64::powi(2., -40));
+/// let var_ks = estimate_keyswitch_noise_lwe_to_glwe_with_constant_term::<u64, _, _, BinaryKeyKind>(
+///     lwe_mask_size,
+///     dispersion_lwe,
+///     dispersion_ks,
+///     base_log,
+///     l_ks,
+/// );
+/// ```
+pub fn estimate_keyswitch_noise_lwe_to_glwe_with_constant_term<T, D1, D2, K>(
+    lwe_mask_size: LweDimension,
+    dispersion_lwe: D1,
+    dispersion_ksk: D2,
+    base_log: DecompositionBaseLog,
+    level: DecompositionLevelCount,
+) -> Variance
+where
+    T: UnsignedInteger,
+    D1: DispersionParameter,
+    D2: DispersionParameter,
+    K: KeyDispersion,
+{
+    let n = lwe_mask_size.0 as f64;
+    let base = (1 << base_log.0) as f64;
+    let q_square = f64::powi(2., (2 * T::BITS) as i32);
+
+    // res 1
+    let res_1 = dispersion_lwe;
+
+    // res 2
+    let res_2 = n
+        * (q_square / (12. * f64::powi(base, 2 * level.0 as i32)) - 1. / 12.)
+        * (K::variance_key_coefficient::<T>().get_modular_variance::<T>()
+            + square(K::expectation_key_coefficient()));
+
+    // res 3
+    let res_3 = n / 4. * K::variance_key_coefficient::<T>().get_modular_variance::<T>();
+
+    // res 4
+    let res_4 =
+        n * (level.0 as f64) * dispersion_ksk.get_modular_variance::<T>() * (square(base) + 2.)
+            / 12.;
+
+    Variance(res_1.get_modular_variance::<T>() + res_2 + res_3 + res_4)
+}
+
+/// Return the variance of the non constant GLWE after an LWE to GLWE keyswitch
+/// # Example
+/// ```rust
+/// use concrete_commons::dispersion::Variance;
+/// use concrete_commons::key_kinds::BinaryKeyKind;
+/// use concrete_commons::parameters::{
+///     DecompositionBaseLog, DecompositionLevelCount, LweDimension,
+/// };
+/// use concrete_npe::estimate_keyswitch_noise_lwe_to_glwe_with_non_constant_terms;
+/// let lwe_mask_size = LweDimension(630);
+/// let l_ks = DecompositionLevelCount(4);
+/// let base_log = DecompositionBaseLog(7);
+/// let dispersion_ks = Variance(f64::powi(2., -40));
+/// // Compute the noise
+/// let var_ks = estimate_keyswitch_noise_lwe_to_glwe_with_non_constant_terms::<u64, _>(
+///     lwe_mask_size,
+///     dispersion_ks,
+///     base_log,
+///     l_ks,
+/// );
+/// ```
+pub fn estimate_keyswitch_noise_lwe_to_glwe_with_non_constant_terms<T, D>(
+    lwe_mask_size: LweDimension,
+    dispersion_ksk: D,
+    base_log: DecompositionBaseLog,
+    level: DecompositionLevelCount,
+) -> Variance
+where
+    T: UnsignedInteger,
+    D: DispersionParameter,
+{
+    let n = lwe_mask_size.0 as f64;
+    let base = (1 << base_log.0) as f64;
+
+    let res =
+        n * (level.0 as f64) * dispersion_ksk.get_modular_variance::<T>() * (square(base) + 2.)
+            / 12.;
+
+    Variance(res)
+}
+/// Returns a variance of U when doing a modulus switching
+/// U is the distribution of error for this bits greater than q.
```

Comment:
There should be a space above the comment. Return + point. Why is that on two lines ? Is anyone supposed to understand what that means ?